### DISCUSSION TITLE
Matcher with large number of patterns

### DISCUSSION THREAD

**Discussion by vblagoje** [explosion/spaCy#10739] [Help: Coding & Implementations] [answered]
Hi, I have a use case where Matcher is initialized with ~800k patterns. It takes about 2 minutes (typical AWS EC2 node) to add all these patterns. Can I somehow serialize the Matcher so that it loads these patterns faster on restart? Anything else I could try? TIA

**Comment by polm** [comment, marked as answer]
This has come up before, see #10514. Can you clarify why two minutes is a problem? That doesn't sound particularly slow for that amount of data if you're only doing it once per process.

Also see #8790, which is about some speed issues in the Matcher we're currently working on, though it's not initialization related.

**Comment by vblagoje** [comment]
@polm, it's not a problem, but the user experience is not the best if a user has to wait that long. I mean, not the end of the world. Just wondering if it could be somehow improved. Thanks for your response and references.
